modules/apps/web-experience/layout/layout-taglib/src/main/resources/META-INF/resources/layouts_tree/js/layouts_tree_state.js

```diff
@@ -280,7 +280,7 @@ AUI.add(
 						node.get('children').forEach(A.bind(instance._restoreCheckedNode, instance));
 					},
 
-					_updateCheckedNodes: function(node, state) {
+					_updateCheckedNodes: function(node, state, force) {
```

Comment:
Basically, we have two scenarios to handle when calling ``_updateCheckedNodes``.

1. Explicit updateCheckedNodes(force equals true or undefined). When click/unclick on a parent node, we expect all its children nodes to be selected/unselected automatically. In this case, we force all children node having consistent states with their parents.

2. Implict updateCheckedNodes(force equals false). When click on load more results or detect childrenNodeChanges, we expect the all local changed children nodes remaining their current status and new loaded children nodes respect their original states. In this case, we allow children nodes and their parents having different states.

The ``state`` is only ``undefined`` when the ``force`` equals true(currently, only line 332).

Due to ``state`` is not defended, we use ``checkedIndex`` to get the node's original state, rather than giving it a state(true or false).